server/src/repositories/telemetry.repository.ts

```diff
@@ -33,20 +48,120 @@ class MetricGroupRepository implements IMetricGroupRepository {
   }
 }
 
+const aggregation = new metrics.ExplicitBucketHistogramAggregation(
+  [0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10, 25, 50, 75, 100, 250, 500, 750, 1000, 2500, 5000, 7500, 10_000],
+  true,
+);
+
+let instance: NodeSDK | undefined;
+
+export const bootstrapTelemetry = (port: number) => {
+  if (instance) {
+    throw new Error('OpenTelemetry SDK already started');
+  }
+  instance = new NodeSDK({
+    resource: new resources.Resource({
+      [SemanticResourceAttributes.SERVICE_NAME]: `immich`,
+      [SemanticResourceAttributes.SERVICE_VERSION]: serverVersion.toString(),
+    }),
+    metricReader: new PrometheusExporter({ port }),
+    contextManager: new AsyncLocalStorageContextManager(),
+    instrumentations: [
+      new HttpInstrumentation(),
+      new IORedisInstrumentation(),
+      new NestInstrumentation(),
+      new PgInstrumentation(),
+    ],
+    views: [new metrics.View({ aggregation, instrumentName: '*', instrumentUnit: 'ms' })],
+  });
+
+  instance.start();
+};
+
+export const teardownTelemetry = async () => {
+  if (instance) {
+    await instance.shutdown();
+    instance = undefined;
+  }
+};
+
 @Injectable()
 export class TelemetryRepository implements ITelemetryRepository {
   api: MetricGroupRepository;
   host: MetricGroupRepository;
   jobs: MetricGroupRepository;
   repo: MetricGroupRepository;
 
-  constructor(metricService: MetricService, @Inject(IConfigRepository) configRepository: IConfigRepository) {
-    const { telemetry } = configRepository.getEnv();
+  constructor(
+    private metricService: MetricService,
+    private reflect: Reflector,
+    @Inject(IConfigRepository) private configRepository: IConfigRepository,
+    @Inject(ILoggerRepository) private logger: ILoggerRepository,
+  ) {
+    const { telemetry } = this.configRepository.getEnv();
     const { apiMetrics, hostMetrics, jobMetrics, repoMetrics } = telemetry;
 
     this.api = new MetricGroupRepository(metricService).configure({ enabled: apiMetrics });
     this.host = new MetricGroupRepository(metricService).configure({ enabled: hostMetrics });
     this.jobs = new MetricGroupRepository(metricService).configure({ enabled: jobMetrics });
     this.repo = new MetricGroupRepository(metricService).configure({ enabled: repoMetrics });
   }
+
+  setup({ repositories }: { repositories: ClassConstructor<unknown>[] }) {
+    const { telemetry } = this.configRepository.getEnv();
+    if (!telemetry.enabled || !telemetry.repoMetrics) {
```

Comment:
This is a slight difference in behavior, right? `IMMICH_METRICS` takes priority over `IMMICH_IO_METRICS` with this change. You mentioned changing this to include/exclude so it doesn't matter much, but if you don't think you'll get to that before the next release then you'd want to update the docs on the new behavior.

Reply:
Enabled gets set to true here if any are enabled., which is how you had it before. I agree it is confusing though.

Reply:
Oh true, I forgot about that too lol. That's fine then